packager/media/formats/mp2t/audio_header.h

```diff
@@ -50,6 +50,8 @@ class AudioHeader {
   /// @return the size of frame (header + payload).
   virtual size_t GetFrameSize() const = 0;
 
+  virtual size_t GetFrameSizeWithoutParsing(const uint8_t* data, size_t num_bytes) const {return 0;}
```

Comment:
Can you add a comment like the functions above? Also define it as pure virtual.

You'll also need to implement the similar function in ac3_header.h.

Reply:
Yeah, this was just a quick-and-dirty implementation to get it to build. I haven't looked at the ac3 side of things. I might ping you if it's not immediately obvious to me how to implement this function for AC3 audio.

Reply:
Ah! Okay, I get was the adts version was doing. Should be able to handle the ac3 version, I think.

Reply:
Since GetFrameSize() actually has to do some logic, would it be bad form to have GetFrameSizeWithoutParsing() actually set fscod_ and frmsizecod_ and then call GetFrameSize() (though, I guess that would make the function non-const)?

Would it be best to have both GetFrameSizes call a common function to do the actual frame size logic?

Or is the AC3 standard fixed enough that duplicating the small amount of logic involved is fine by y'all?

Reply:
Like, does this work for you?

```
/// Calculate the size of the frame from the sample rate code and the 
/// frame size code.
/// @return the size of the frame (header + payload).
size_t CalcFrameSize(uint8_t fscod, uint8_t frmsizecod) {
  const size_t kNumFscode = arraysize(kAc3SampleRateTable);
  DCHECK_LT(fscod, kNumFscode);
  DCHECK_LT(frmsizecod, arraysize(kFrameSizeCodeTable));
  // The order of frequencies are reversed in |kFrameSizeCodeTable| compared to
  // |kAc3SampleRateTable|.
  const int index = kNumFscode - 1 - fscod;
  return kFrameSizeCodeTable[frmsizecod][index] * 2;
}

size_t Ac3Header::GetFrameSize() const {
  return CalcFrameSize(fscod_, frmsizecod_);
}

size_t Ac3Header::GetFrameSizeWithoutParsing(const uint8_t* data, size_t num_bytes) const {
  uint8_t fscod = data[5] & 0xc0;
  uint8_t frmsizecod = data[5] & 0x3f;
  return CalcFrameSize(fscod, frmsizecod);
}
```

Reply:
Yes, that looks very good to me!

> uint8_t fscod = data[5] & 0xc0;

Should be data[4] instead of data[5].

Also, there should be check of 'num_bytes' to make sure that it is >4.

Reply:
Good catch! I hadn't actually tested it yet. 'preciate it!